### PR TITLE
MAT-6852 QDM - Make Search work in Code sub-tab on CQL Editor screen

### PR DESCRIPTION
**Pull request by adongare**
## MADiE PR

Jira Ticket: [MAT-6852](https://jira.cms.gov/browse/MAT-6852)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included
* [x] No extemporaneous files are included (i.e Complied files or testing results)
* [x] This PR is in to the **correct branch**.
* [ ] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [ ] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependancies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

https://github.com/MeasureAuthoringTool/madie-editor/assets/23436747/80007a1d-8833-4e0a-b9b1-136b37210564

### CODE REVIEW

File: src/terminologySectionPanel/codesSection/codesSubSection/codeSubSection/CodeSubSection.tsx [hunk 1]
```diff
@@ -13,9 +17,40 @@ export default function CodeSubSection({
...
+  const [toastOpen, setToastOpen] = useState<boolean>(false);
+  const [toastMessage, setToastMessage] = useState<string>("");
+  const onToastClose = () => {
+    setToastMessage("");
+    setToastOpen(false);
+  };
+
+  const handleFormSubmit = async (values) => {
+    // eslint-disable-next-line
+    const terminologyService = await useTerminologyServiceApi();
```
Comment (adongare) [inline src/terminologySectionPanel/codesSection/codesSubSection/codeSubSection/CodeSubSection.tsx:32]: this hook needs some refactoring. it makes requests to get service config each time we use this hook.

**Review by sb-cecilialiu — DISMISSED** [review]
Looks good to me.

File: src/terminologySectionPanel/codesSection/codesSubSection/codeSubSection/CodeSubSection.tsx [hunk 2]
```diff
@@ -13,9 +17,40 @@ export default function CodeSubSection({
...
+      try {
+        const code = await terminologyService.getCodeDetails(
+          values.code,
+          values.codeSystemName,
+          values.version
+        );
+        setCode(code);
+      } catch (error) {
+        if (error.response?.status === 404) {
+          setCode(undefined);
```
Comment (RohitKandimalla) [inline src/terminologySectionPanel/codesSection/codesSubSection/codeSubSection/CodeSubSection.tsx]: Should we display any error to the user ?

File: src/terminologySectionPanel/codesSection/codesSubSection/codeSubSection/CodeSubSection.tsx [hunk 3]
```diff
@@ -13,9 +17,40 @@ export default function CodeSubSection({
...
+        } else {
+          console.error(error);
+          setToastMessage(
+            "An issue occurred while retrieving the code from VSAC. Please try again. If the issue continues, please contact helpdesk."
+          );
+          setToastOpen(true);
+        }
+      }
+      setShowResultsTable(true);
```
Comment (RohitKandimalla) [inline src/terminologySectionPanel/codesSection/codesSubSection/codeSubSection/CodeSubSection.tsx]: Should we display results table only on successful API call ?

File: src/api/useTerminologyServiceApi.ts [hunk 4]
```diff
@@ -136,6 +144,20 @@ export class TerminologyServiceApi {
...
+    codeSystem: string,
+    version: string
+  ): Promise<Code> {
+    const response = await axios.get(`${this.baseUrl}/terminology/code`, {
+      params: { code: code, codeSystem: codeSystem, version: version },
+      headers: {
+        Authorization: `Bearer ${this.getAccessToken()}`,
+      },
+    });
+    return response.data;
```
Comment (RohitKandimalla) [inline src/api/useTerminologyServiceApi.ts]: I have to test and see what will happen when a api call fails, but looks like we are awaiting for async/promise to resolve but we are not handling it.

So I was wondering if we can just return the promise the axios.get() and also remove async for this method

Comment (RohitKandimalla) [reply src/api/useTerminologyServiceApi.ts]:
```
  getCodeDetails(
    code: string,
    codeSystem: string,
    version: string
  ): Promise<any> {
    return axios.get(`${this.baseUrl}/terminology/code`, {
      params: { code: code, codeSystem: codeSystem, version: version },
      headers: {
        Authorization: `Bearer ${this.getAccessToken()}`,
      },
    });
  }
```

File: src/terminologySectionPanel/codesSection/codesSubSection/codeSubSection/CodeSubSection.tsx (hunk 3 above)
Comment (RohitKandimalla) [reply src/terminologySectionPanel/codesSection/codesSubSection/codeSubSection/CodeSubSection.tsx:52]: I saw your video good to know that there is a message already in the empty table/

File: src/terminologySectionPanel/codesSection/codesSubSection/codeSubSection/CodeSubSection.tsx (hunk 2 above)
Comment (RohitKandimalla) [reply src/terminologySectionPanel/codesSection/codesSubSection/codeSubSection/CodeSubSection.tsx]: I saw your video good to know that there is a message already in the empty table/

Comment (adongare) [reply src/terminologySectionPanel/codesSection/codesSubSection/codeSubSection/CodeSubSection.tsx]: yeah, 404 does not need an error message. we display code not found message in table

File: src/terminologySectionPanel/codesSection/codesSubSection/codeSubSection/CodeSubSection.tsx (hunk 3 above)
Comment (adongare) [reply src/terminologySectionPanel/codesSection/codesSubSection/codeSubSection/CodeSubSection.tsx]: this flag is misleading. this is not really to display a table but to expand/collapse the result section. my take on this is to show result section in expanded state all the time unless the user clicks manually on the caret icon to expand/collapse. in that case we do not even need this flag. bu this is something Michael to decide

File: src/api/useTerminologyServiceApi.ts (hunk 4 above)
Comment (adongare) [reply src/api/useTerminologyServiceApi.ts]: tested...it works. thank you!

**Review by RohitKandimalla — DISMISSED** [review]

**Review by RohitKandimalla — APPROVED** [review]

**Pull request merged**